jupyter_console/ptshell.py

```diff
@@ -350,8 +379,11 @@ def init_prompt_toolkit_cli(self):
         if self.simple_prompt or ('JUPYTER_CONSOLE_TEST' in os.environ):
             # Simple restricted interface for tests so we can find prompts with
             # pexpect. Multi-line input not supported.
+            @asyncio.coroutine
             def prompt():
```

Comment:
How does this affect prior use of `prompt` that wasn't expecting `asyncio`?

Reply:
Thk for fast and good review : the spot the nick
I understand you are speaking of old version of `prompt_toolkit` ?
On `prompt-toolkit==2.0.1` (the oldest). It works smoothness BUT: There is an ugly error message. I realized it is not due to this PR (same in master). I paste below

Reply:
Here is the error message (due to history apparently) after
```bash
pip uninstall prompt-toolkit
pip install prompt-toolkit==2.0.1
```


```text
Jupyter console 6.1.1dev

Python 3.7.5 (default, Nov 20 2019, 09:21:52)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.10.1 -- An enhanced Interactive Python. Type '?' for help.
Future exception was never retrieved
future: <prompt_toolkit.eventloop.future.Future object at 0x7f1e4501bf50>
Traceback (most recent call last):
  File "/home2/tourneboeuf/.local/lib/python3.7/site-packages/prompt_toolkit/eventloop/async_generator.py", line 113, in consume_async_generator
    item = iterator.send(send)
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home2/tourneboeuf/.local/lib/python3.7/site-packages/prompt_toolkit/eventloop/coroutine.py", line 90, in step_next
    new_f = coroutine.throw(exc)
  File "/home2/tourneboeuf/.local/lib/python3.7/site-packages/prompt_toolkit/history.py", line 57, in _start_loading
    item_callback=add_string))
  File "/home2/tourneboeuf/.local/lib/python3.7/site-packages/prompt_toolkit/eventloop/coroutine.py", line 86, in step_next
    new_f = coroutine.send(None)
RuntimeError: generator raised StopIteration
Future exception was never retrieved
future: <prompt_toolkit.eventloop.future.Future object at 0x7f1e4501ebd0>
Traceback (most recent call last):
  File "/home2/tourneboeuf/.local/lib/python3.7/site-packages/prompt_toolkit/eventloop/async_generator.py", line 113, in consume_async_generator
    item = iterator.send(send)
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home2/tourneboeuf/.local/lib/python3.7/site-packages/prompt_toolkit/eventloop/coroutine.py", line 90, in step_next
    new_f = coroutine.throw(exc)
  File "/home2/tourneboeuf/.local/lib/python3.7/site-packages/prompt_toolkit/history.py", line 57, in _start_loading
    item_callback=add_string))
  File "/home2/tourneboeuf/.local/lib/python3.7/site-packages/prompt_toolkit/eventloop/coroutine.py", line 86, in step_next
    new_f = coroutine.send(None)
RuntimeError: generator raised StopIteration
Future exception was never retrieved
future: <prompt_toolkit.eventloop.future.Future object at 0x7f1e44fac550>
Traceback (most recent call last):
  File "/home2/tourneboeuf/.local/lib/python3.7/site-packages/prompt_toolkit/eventloop/async_generator.py", line 113, in consume_async_generator
    item = iterator.send(send)
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home2/tourneboeuf/.local/lib/python3.7/site-packages/prompt_toolkit/eventloop/coroutine.py", line 90, in step_next
    new_f = coroutine.throw(exc)
  File "/home2/tourneboeuf/.local/lib/python3.7/site-packages/prompt_toolkit/history.py", line 57, in _start_loading
    item_callback=add_string))
  File "/home2/tourneboeuf/.local/lib/python3.7/site-packages/prompt_toolkit/eventloop/coroutine.py", line 86, in step_next
    new_f = coroutine.send(None)
RuntimeError: generator raised StopIteration

In [1]:
```